src/ring_theory/power_series/well_known.lean

```diff
@@ -83,4 +84,74 @@ variables {A A'} (n : ℕ) (f : A →+* A')
+  map_one' := by { ext1, simp only [mul_boole, coeff_mk, coeff_one], split_ifs,
+                rw [h, pow_zero], refl, },
+  map_add' := by {intros, ext, norm_num, rw mul_add, },
+  map_mul' := by {intros, ext, rw [coeff_mul, coeff_mk, coeff_mul, finset.mul_sum],
+                apply sum_congr rfl, norm_num, intros b c H, rw [<-H, pow_add], ring, }, }
+
+/-- Shows that `e^(aX) * e^(bX) = e^((a + b)X) ` -/
+theorem exp_mul_exp_eq_exp_add [algebra ℚ A] (a b : A) :
+  eval_mul_hom a (exp A) * eval_mul_hom b (exp A) = eval_mul_hom (a + b) (exp A) :=
+begin
+  ext, rw [coeff_mul, exp, eval_mul_hom, eval_mul_hom, eval_mul_hom],
+  simp only [coeff_mk, coe_mk, factorial], rw nat.sum_antidiagonal_eq_sum_range_succ_mk,
+  simp only [factorial], rw [add_pow, sum_mul], apply sum_congr, { refl },
+  { rintros x hx,
+    rw [mul_assoc (a^x * b ^ (n - x)) _ _, mul_assoc (a^x) _ _, ←mul_assoc _  (b ^ (n - x)) _,
+    mul_comm _ (b^(n - x)), ←mul_assoc (a^x) _ _, ←mul_assoc (a^x) _ _],
+    suffices f : (algebra_map ℚ A) (1 / ↑(x.factorial)) * (algebra_map ℚ A)
+     (1 / ↑((n - x).factorial)) = (↑(n.choose x) * (algebra_map ℚ A) (1 / ↑(n.factorial))),
+      { rw [←f, mul_assoc], },
+    rw [←map_nat_cast (algebra_map ℚ A) (n.choose x), ←map_mul, ←map_mul],
+    refine ring_hom.congr_arg _ _, rw [mul_one_div ↑(n.choose x) _, one_div_mul_one_div],
+    symmetry, rw [div_eq_iff, div_mul_eq_mul_div, one_mul, choose_eq_factorial_div_factorial],
+    norm_cast, rw cast_dvd_char_zero,
+    { apply factorial_mul_factorial_dvd_factorial (mem_range_succ_iff.1 hx), },
+    { apply mem_range_succ_iff.1 hx, },
+    { rintros h, apply factorial_ne_zero n, rw cast_eq_zero.1 h, },
+  },
+end
+
+lemma eval_mul_hom_zero [algebra ℚ A] (f : power_series A) :
+  eval_mul_hom 0 f = C A (constant_coeff A f) :=
+begin
+  rw [eval_mul_hom, coe_mk], ext, rw [coeff_mk, power_series.coeff_C], split_ifs,
+  { rw h, simp only [one_mul, coeff_zero_eq_constant_coeff, pow_zero], },
+  { rw [zero_pow' n h, zero_mul], },
+end
+
+lemma eval_mul_hom_one [algebra ℚ A] (f : power_series A) :
+  eval_mul_hom 1 f = f :=
+by { rw eval_mul_hom, ext, simp only [one_pow, coeff_mk, one_mul, coe_mk], }
+
+/-- The ring homomorphism taking a power series `f(X)` to `f(aX)`. -/
+noncomputable def eval_neg_hom : power_series A →+* power_series A :=
+eval_mul_hom (-1 : A)
+
+/-- Shows that `e^{x} * e^{-x} = 1` -/
+theorem exp_mul_exp_neg_eq_one [algebra ℚ A] : exp A * eval_neg_hom (exp A) = 1 :=
+begin
+  rw eval_neg_hom,
+  conv_lhs { congr, rw ←eval_mul_hom_one (exp A), },
+  rw exp_mul_exp_eq_exp_add, simp, rw eval_mul_hom_zero, simp,
+end
+
+@[simp] lemma eval_neg_hom_X : @eval_neg_hom A _ X = -X :=
+begin
+  rw eval_neg_hom, ext, simp only [linear_map.map_neg], rw coeff_X, split_ifs,
+  { rw [h, eval_mul_hom], simp only [coeff_mk, mul_one, coe_mk, coeff_one_X, pow_one], },
+  { rw eval_mul_hom, simp, suffices f : (coeff A n) X = 0, {rw f, rw mul_zero,},
+    rw coeff_X, split_ifs, refl, },
+end
```

Comment:
The defs and lemmas here which are not specific to `exp` belong in `ring_theory/power_series/basic`

Reply:
Done, thank you!